### PR TITLE
Fix: bug when nspin = 2, fixed smearing

### LINKED ISSUE #3073: Confusing population results calculated on di-Sulfur (S2) system
### Describe the bug

only switch off smearing, total number of electrons are turned out to be different in `running_scf.log` file  
with smearing:  
image
without smearing:  
image

### Expected behavior

_No response_

### To Reproduce

[disulfur.zip](https://github.com/deepmodeling/abacus-develop/files/12927514/disulfur.zip)

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers

- [X] Verify the issue is not a duplicate.
- [X] Describe the bug.
- [X] Steps to reproduce.
- [X] Expected behavior.
- [X] Error message.
- [X] Environment details.
- [X] Additional context.
- [x] Assign a priority level (low, medium, high, urgent).
- [x] Assign the issue to a team member.
- [x] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

### PR DESCRIPTION
**Pull request by Qianruipku**
fix issue #3073 
Reference QE code:
image

### CODE REVIEW

File: source/module_elecstate/occupy.cpp [hunk 1]
```diff
@@ -131,8 +131,12 @@ void Occupy::iweights(const int nks,
                       const std::vector<int>& isk)
 {
     assert(is < 2);
+    double degspin = 2.0;
+    if (GlobalV::NSPIN == 4)
+        degspin = 1.0;
+    if (is != -1)
```
Comment (dyzheng) [inline source/module_elecstate/occupy.cpp:137]: Sorry, I think I didn't understand the difference between new code and old code.
The judgement of "is != -1" is not same as "GlobalV::NSPIN == 2"?

**Review by dyzheng — CHANGES_REQUESTED** [review]
Please fix the UT: elecstate_occupy

File: source/module_elecstate/occupy.cpp (hunk 1 above)
Comment (Qianruipku) [reply source/module_elecstate/occupy.cpp:137]: When use TWO_EFERMI, is = 0 or 1.  When not use TWO_EFERMI， is = -1。
To be honest，I am not familiar with TWO_EFERMI and I just copy the code of QE.

**Review by dyzheng — APPROVED** [review]

**Review by dyzheng — COMMENTED** [review]
This PR fixed the fixed smearing bug of TWO_EFERMI case which can be triggered by parameter "nupdown". This case is rarely used because NSPIN=2 usually set with metal or semiconductor structures.

**Pull request merged**